streams/src/main/java/org/apache/kafka/streams/processor/internals/PunctuationSchedule.java

```diff
@@ -27,8 +27,8 @@
     // this Cancellable will be re-pointed at the successor schedule in next()
     private final RepointableCancellable cancellable;
 
-    PunctuationSchedule(ProcessorNode node, long interval, Punctuator punctuator) {
-        this(node, 0L, interval, punctuator, new RepointableCancellable());
+    PunctuationSchedule(ProcessorNode node, long time, long interval, Punctuator punctuator) {
```

Comment:
Nit: can you improve this further by adding `final` to all parameters and reformat to single parameter per line formatting?